### PR TITLE
[quantization] SmoothQuant support for vision components of Qwen3-vl

### PR DESCRIPTION
**Pull request by Torrero**
This PR supports SmoothQuant for vision components of Qwen3-vl

It allows to reduce accuracy degradation by 3%. Tested for `vqav2` dataset.

SmoothQuant is appllied for `vision_block` and for `vision_patch_merger`

|model|vqav2| accuracy degradation|
|--|--|--|
|original|0.895| - |
|quantized W4A16 + GPTQ + SMSE |0.695| 20%|
|quantized W4A16 + GPTQ + SMSE + smoothquant (alpha=0.8)|0.721| 17% |

Run command:
```
python tico/quantization/wrapq/examples/quantize_qwen3_vl_with_gptq.py --model=Qwen/Qwen3-VL-4B-Instruct --cache_dir=<YOUR_CACHE_DIR> --sensitivity_path=<YOUR_PATH> --trust-remote-code --eval_tasks=vqav2 --gptq_mse=smse --nsamples_for_evaluation=1000 --embedding_weight_bits=8 --vision_patch_embed_weight_bits=4 --linear_weight_bits=4 --lm_head_weight_bits=4 --nsamples_for_qcalibration=128 --smoothquant --smoothquant_alpha=0.8
```

TICO-DCO-1.0-Signed-off-by:  Evgenii Maltsev <e.maltsev@samsung.com>

### CODE REVIEW

File: tico/quantization/algorithm/smoothquant/smooth_quant.py [hunk 1]
```diff
@@ -280,11 +282,170 @@ def _apply_if_fairseq_relu_bridge(
...
+                if isinstance(attr, torch.nn.Linear):
+                    back_modules_attn.append(attr)
+        if not back_modules_attn:
+            return False
+
+    # Smooth norm1 → attention
+    attn_key = f"{name}.attn"
+    # Find the first linear's input key for activation stats
+    first_linear = back_modules_attn[0]
+    for key_suffix in ["", ".qkv", ".q_proj", ".q", ".query"]:
```
Comment (mhs4670go) [inline tico/quantization/algorithm/smoothquant/smooth_quant.py:344]: Seems that current codes return `True` even though there's missing keys. How about this one?

```python
applied_attn = False

for key_suffix in ["", ".qkv", ".q_proj", ".q", ".query"]:
    potential_key = f"{attn_key}{key_suffix}"
    if potential_key in activation_max:
        smooth_weights(
            module.norm1,
            back_modules_attn,
            activation_max[potential_key],
            alpha_to_apply,
        )
        applied_attn = True
        break

if not applied_attn:
    print(
        f"[SmoothQuant] Warning: activation stats not found for "
        f"{name} attention input."
    )

```

**Comment by mhs4670go** [comment]
@Torrero LM decoder part will be covered as well later?

**Comment by Torrero** [comment]
> @Torrero LM decoder part will be covered as well later?

Yes, I will support it in the next PR.

File: tico/quantization/algorithm/smoothquant/smooth_quant.py [hunk 2]
```diff
@@ -280,11 +282,188 @@ def _apply_if_fairseq_relu_bridge(
...
+        return False
+
+    # Smooth norm → linear_fc1
+    fc1_key = f"{name}.linear_fc1"
+    if fc1_key in activation_max:
+        fc1_input_scales = activation_max[fc1_key]
+        smooth_weights(
+            module.norm, [module.linear_fc1], fc1_input_scales, alpha_to_apply
+        )
+        return True
```
Comment (mhs4670go) [inline tico/quantization/algorithm/smoothquant/smooth_quant.py:455]: How about applying same logic here, too? I want them not to skip the smoothing quietly.

```suggestion
    if fc1_key in activation_max:
        fc1_input_scales = activation_max[fc1_key]
        smooth_weights(
            module.norm, [module.linear_fc1], fc1_input_scales, alpha_to_apply
        )
        return True
    else:
        # print warning
```

File: tico/quantization/algorithm/smoothquant/smooth_quant.py (hunk 1 above)
Comment (Torrero) [reply tico/quantization/algorithm/smoothquant/smooth_quant.py:344]: Thank you, I changed it

File: tico/quantization/algorithm/smoothquant/smooth_quant.py (hunk 2 above)
Comment (Torrero) [reply tico/quantization/algorithm/smoothquant/smooth_quant.py:455]: Good point, I fixed it, thank you!

**Comment by mhs4670go** [comment]
LGTM

**Review by mhs4670go — APPROVED** [review]
